container-disc/src/main/java/com/yahoo/container/jdisc/component/Deconstructor.java

```diff
@@ -52,27 +55,44 @@ public void deconstruct(Object component) {
     }
 
     private static class DestructComponentTask implements Runnable {
+
         private final AbstractComponent component;
 
         DestructComponentTask(AbstractComponent component) {
             this.component = component;
         }
 
+        /** Returns a random value which will be different across identical containers invoking this at the same time */
+        private long random() {
+            return new SecureRandom().nextLong();
+        }
+
         public void run() {
             log.info("Starting deconstruction of " + component);
             try {
                 component.deconstruct();
                 log.info("Finished deconstructing " + component);
-            } catch (Error e) {
-                try {
-                    Thread.sleep((long) (new Random(System.nanoTime()).nextDouble() * 180 * 1000));
-                } catch (InterruptedException exception) { }
-                com.yahoo.protect.Process.logAndDie("Error when deconstructing " + component, e);
-            } catch (Exception e) {
+            }
+            catch (Exception | NoClassDefFoundError e) { // May get class not found due to it being already unloaded
```

Comment:
This is a resource leak, as the component might not have freed it's resources. However, including NoClassDef error doesn't make it that much worse. But it's something to look for if the container runs out of memory.

Reply:
Agree on both points.